src/lib.rs

```diff
@@ -229,11 +284,14 @@ impl AllocList {
 
     /// Updates the size associated with a base pointer (perfomed on a realloc).
     /// This must not be called with an inner pointer.
-    fn update(&mut self, ptr: usize, size: usize) {
+    fn update(&mut self, ptr: usize, new_ptr: usize, size: usize) {
         for block in self.iter_mut() {
```

Comment:
Possible optimisation for later: store allocations sorted so that locating the correct pointer can be log time using a chop?

Reply:
I considered it, but I actually care about faster insertion than lookup. Storing allocations sorted would have worst case O(n^2). Insertion happens on every single `alloc` call, whereas lookup will happen only during the STW phase of GC. 

Arguably it makes sense to try and reduce the STW pause in the future, but at a cost of slowing down every allocation is, I think, the wrong trade-off.